src/main/java/lotto/LottoProgram.java

```diff
@@ -1,26 +1,36 @@
 package lotto;
 
+import static lotto.view.InputView.*;
 import static lotto.view.InputView.inputBonusNumber;
+import static lotto.view.InputView.inputManualLottoCount;
 import static lotto.view.InputView.inputPurchaseAmount;
 import static lotto.view.InputView.inputWinningNumbers;
 import static lotto.view.ResultView.printLottosCount;
+import static lotto.view.ResultView.printPurchaseLottoByEachType;
 import static lotto.view.ResultView.printReturnRate;
 import static lotto.view.ResultView.printWinningResult;
 
 public class LottoProgram {
 
     public void run() {
-        Money purchaseAmount = new Money(inputPurchaseAmount());
+        Money money = new Money(inputPurchaseAmount());
+        PurchaseAmount amount = PurchaseAmount.of(money.purchase(), inputManualLottoCount());
 
-        Lottos lottos = Lottos.create(
-            purchaseAmount, new RandomNumberGenerator());
-        printLottosCount(purchaseAmount.purchase(), lottos);
+        Lottos lottos = buyLotto(amount);
+        printPurchaseLottoByEachType(amount);
+        printLottosCount(lottos);
```

Comment:
```suggestion
        amount.buyManualLotto(inputManualLottoNumbers(amount.getManual()));
        amount.buyAutoLotto();
        printPurchaseLottoByEachType(amount);
        printLottosCount(amount.allLotto());
```

저는 이런생각을 해봤어요.

- 구매금액보다 많은 로또를 구매하지 못하게 하는 비즈니스 규칙을 잘 관리할 수 있게 하는방법은 무엇을까?
- 수동 구매한 로또 갯수와 , 자동 구매 갯수의 정합성을 항상 맞아야 할꺼같은데, 그렇다면  amount , manualLotto, autoLotto 가 같이 관리하는게 비즈니스 정합성을 유지하기 편하지 않을까?
- 도메인을 따로 관리한다면, 중요한 비즈니스 규칙이 서비스 레이어에 흩어져 있지 않을까? 그렇다면 이는 확장성이나 유지보수하기 힘들지 않을까?

다양한게 고민해보아요~ 😄

Reply:
구매금액보다 많은 로또를 구매하지 못하는 비즈니스 규칙은 구현하지 않았지만,
Customer라는 관리 객체를 만들면 좀 더 비즈니스 규칙이 관리하기가 확실히 쉬울 것 같네요!!